lhotse/audio.py

```diff
@@ -315,7 +315,7 @@ class Recording:
     @staticmethod
     def from_file(
         path: Pathlike,
-        recording_id: Optional[str] = None,
+        recording_id: Optional[Union[str, Callable[[Pathlike], str]]] = None,
```

Comment:
```suggestion
        recording_id: Optional[Union[str, Callable[[Path], str]]] = None,
```

looking at the code, it will always be of type `Path`